src/donor_dashboard/views.py

```diff
@@ -73,14 +73,15 @@ def manage_organization(request, organization_id):
 
 @login_required
 def delete_organization(request, organization_id):
+    if request.method == "POST":
+        organization = Organization.objects.get(organization_id=organization_id)
+        organization_name = organization.organization_name
+        organization.delete()
```

Comment:
I think instead of deleting permanently, we would have to change the state to inactive so that any donations/orders/reservations already mapped to this organization can remain without breaking.